smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,687 @@
+        """Set the launcher for the experiment"""
+        if launcher_type != self._launcher_type:
+            self._launcher_type = launcher_type
+            self._launcher = self.init_launcher(launcher_type)
+            self.job_manager.set_launcher(self._launcher)
+            self.job_manager.add_job_onstep_callback(track_timestep)
+
+            self.job_manager.start()
+
+    @property
+    def launcher(self) -> Launcher:
+        """Return a launcher appropriate for the experiment"""
+        if not self._launcher:
+            self._launcher = LocalLauncher()
+
+        return self._launcher
+
+    def process_manifest(self, manifest_path: str) -> None:
+        """Read the runtime manifest for the experiment and track new entities
+
+        :param manifest_path: The full path to the manifest file
+        :type manifest_path: str"""
+        try:
+            manifest = load_manifest(manifest_path)
+            if not manifest:
+                return
+        except json.JSONDecodeError:
+            self._logger.error(f"Malformed manifest encountered: {manifest_path}")
+            return
+        except ValueError:
+            self._logger.error("Manifest content error", exc_info=True)
+            return
+
+        self.set_launcher(manifest.launcher)
+
+        if not self._launcher:
+            raise SmartSimError(f"Unable to set launcher from {manifest_path}")
+
+        runs = [run for run in manifest.runs if run.timestamp not in self._tracked_runs]
+
+        exp_dir = pathlib.Path(manifest_path).parent.parent.parent
+
+        for run in runs:
+            for entity in run.flatten(
+                filter_fn=lambda e: e.key not in self._tracked_jobs and e.is_managed
+            ):
+                entity.path = str(exp_dir)
+
+                self._tracked_jobs[entity.key] = entity
+                track_event(
+                    run.timestamp,
+                    entity.task_id,
+                    entity.step_id,
+                    entity.type,
+                    "start",
+                    pathlib.Path(entity.status_dir),
+                    self._logger,
+                )
+
+                if entity.is_managed:
+                    self.job_manager.add_job(
+                        entity.name,
+                        entity.task_id,
+                        entity,
+                        False,
+                    )
+                    self._launcher.step_mapping.add(
+                        entity.name, entity.step_id, entity.task_id, True
+                    )
+            self._tracked_runs[run.timestamp] = run
+
+    def on_modified(self, event: FileModifiedEvent) -> None:
+        """Event handler for when a file or directory is modified.
+
+        :param event: Event representing file/directory modification.
+        :type event: FileModifiedEvent"""
+        super().on_modified(event)  # type: ignore
+        print(f"processing modified manifest @ {event.src_path}")
```

Comment:
super pedantic "can we replace `print` with `logger`" request here

Reply:
debug code has feelings too!